### PR TITLE
Adding snowflake as a bundled database secrets plugin

### PR DESCRIPTION
**Pull request by Valarissa**
This PR seeks to add the (Snowflake db secrets plugin)[https://github.com/hashicorp/vault-plugin-database-snowflake] as a bundled plugin.

It is targeted for release with Vault 1.7

### CODE REVIEW

File: helper/builtinplugins/registry.go [hunk 1]
```diff
@@ -95,11 +94,10 @@ func newRegistry() *registry {
 		databasePlugins: map[string]BuiltinFactory{
 			// These four plugins all use the same mysql implementation but with
 			// different username settings passed by the constructor.
-			"mysql-database-plugin":        dbMysql.New(dbMysql.MetadataLen, dbMysql.MetadataLen, dbMysql.UsernameLen),
-			"mysql-aurora-database-plugin": dbMysql.New(credsutil.NoneLength, dbMysql.LegacyMetadataLen, dbMysql.LegacyUsernameLen),
-			"mysql-rds-database-plugin":    dbMysql.New(credsutil.NoneLength, dbMysql.LegacyMetadataLen, dbMysql.LegacyUsernameLen),
-			"mysql-legacy-database-plugin": dbMysql.New(credsutil.NoneLength, dbMysql.LegacyMetadataLen, dbMysql.LegacyUsernameLen),
-
```
Comment (calvn) [inline helper/builtinplugins/registry.go:102]: I think this newline was intentional.

File: vault/plugin_catalog_test.go [hunk 2]
```diff
@@ -129,6 +130,8 @@ func TestPluginCatalog_List(t *testing.T) {
 	}
 
 	if len(plugins) != len(builtinKeys) {
+		log.Printf("%s", builtinKeys)
```
Comment (calvn) [inline vault/plugin_catalog_test.go:133]: Looks like debug logs. If so, can we remove these?

File: vault/testing.go [hunk 3]
```diff
@@ -2078,7 +2078,6 @@ func (m *mockBuiltinRegistry) Keys(pluginType consts.PluginType) []string {
 		"mysql-aurora-database-plugin",
 		"mysql-rds-database-plugin",
 		"mysql-legacy-database-plugin",
-
```
Comment (calvn) [inline vault/testing.go:2081]: This newline might have been intentional.

File: helper/builtinplugins/registry.go (hunk 1 above)
Comment (Valarissa) [reply helper/builtinplugins/registry.go]: Re-added

File: vault/plugin_catalog_test.go (hunk 2 above)
Comment (Valarissa) [reply vault/plugin_catalog_test.go]: Removed

File: vault/testing.go (hunk 3 above)
Comment (Valarissa) [reply vault/testing.go]: Re-added

File: vault/plugin_catalog_test.go [hunk 4]
```diff
@@ -3,6 +3,8 @@ package vault
 import (
 	"context"
 	"fmt"
+	"github.com/hashicorp/vault/sdk/helper/consts"
```
Comment (calvn) [inline vault/plugin_catalog_test.go:6]: The import here seems to be off. Can you revert this?

**Review by calvn — APPROVED** [review]
One last tiny thing, otherwise LGTM!

**Pull request merged**